### PR TITLE
fix(slack): Allow issue links to unfurl properly with block kit

### PR DESCRIPTION
**Pull request by isabellaenriquez**
Fixes an issue where links to Sentry issues posted in Slack wouldn't unfurl when block kit is enabled. Also fixes an issue with some linked issue action payloads being malformed (unresolved, mark as ongoing, assignment would do the action in Sentry but would not successfully show that it worked in Slack if a user used any one of these actions on an unfurled Sentry issue link in Slack). Finally, this PR also adds much-needed tests for using Slack actions through unfurled Sentry links when block kit is enabled.

Fixes (some of) SENTRY-2DHB

image

### CODE REVIEW

File: src/sentry/integrations/slack/requests/action.py [hunk 1]
```diff
@@ -46,6 +46,8 @@ def callback_data(self) -> JSONData:
         if self.data["type"] == "block_actions":
             if self.data.get("view"):
                 return json.loads(self.data["view"]["private_metadata"])
+            elif self.data["container"].get("is_app_unfurl"):
+                return json.loads(self.data["app_unfurl"]["blocks"][0]["block_id"])
```
Comment (isabellaenriquez) [inline src/sentry/integrations/slack/requests/action.py:50]: this is used to carry out actions on unfurled issues in slack; i will try to add tests for this but i don't believe there are any existing ones for when block kit isn't enabled

File: src/sentry/integrations/slack/requests/options_load.py [hunk 2]
```diff
@@ -18,6 +18,8 @@ class SlackOptionsLoadRequest(SlackRequest):
 
     @property
     def group_id(self) -> int:
+        if self.data["container"].get("is_app_unfurl"):
+            return int(json.loads(self.data["app_unfurl"]["blocks"][0]["block_id"])["issue"])
```
Comment (isabellaenriquez) [inline src/sentry/integrations/slack/requests/options_load.py:22]: for carrying out assignment on unfurl issues in slack, see note above about testing

File: src/sentry/integrations/slack/requests/action.py (hunk 1 above)
Comment (scefali) [inline src/sentry/integrations/slack/requests/action.py:50]: @isabellaenriquez feels like we should check for an empty array here so this doesn't blow up

File: src/sentry/integrations/slack/webhooks/event.py [hunk 3]
```diff
@@ -179,6 +179,11 @@ def on_link_shared(self, request: Request, slack_request: SlackDMRequest) -> boo
         if not results:
             return False
 
+        if features.has("organizations:slack-block-kit", organization):
+            for link_info in results.values():
+                if "text" in link_info:
+                    del link_info["text"]
```
Comment (scefali) [inline src/sentry/integrations/slack/webhooks/event.py:185]: @isabellaenriquez a comment explaining why we have to delete this field would be nice

File: src/sentry/integrations/slack/requests/action.py (hunk 1 above)
Comment (isabellaenriquez) [reply src/sentry/integrations/slack/requests/action.py:50]: you mean just checking that there are blocks in `"app_unfurl"`?

Comment (scefali) [reply src/sentry/integrations/slack/requests/action.py:50]: that and `data["app_unfurl"]["blocks"]` is an array of at least size 1

File: src/sentry/integrations/slack/requests/options_load.py (hunk 2 above)
Comment (ceorourke) [reply src/sentry/integrations/slack/requests/options_load.py:22]: Is `container` guaranteed? You could do `self.data.get("container", {}).get("is_app_unfurl")` to be safe

**Review by ceorourke — APPROVED** [review]

**Pull request merged**